macros/coalesce_fields.sql

```diff
@@ -75,11 +75,11 @@
                 {#- handle booleans with especial care -#}
                 {%- if column.datatype in ('boolean', 'bo') %}
                 case
-                    when {{column.name}} = true then cast('true' as {{dbt_utils.type_string()}})
-                    when {{column.name}} = false then cast('false' as {{dbt_utils.type_string()}})
+                    when {{column.name}} = true then cast('true' as {{type_string()}})
```

Comment:
To support the switch from `dbt_utils.type_string()` &rarr; just `type_string()`, we should also bump the `require-dbt-version` in this package to `>=1.2.0`, since that's when the data type macros were added to `dbt-core` / adapters